webpack.mix.js

```diff
@@ -12,6 +12,8 @@ const { mix } = require('laravel-mix');
  */
 
 mix.js('resources/assets/js/app.js', 'public/build/js')
-    .sass('resources/assets/sass/app.scss', 'public/build/css')
-    .version()
-    .copy('node_modules/markdown/lib/markdown.js', 'public/build/custom/markdown.js');
+    .sass('resources/assets/sass/app.scss', 'public/build/css');
+
+if (mix.config.inProduction) {
+    mix.version();
+}
```

Comment:
Why wouldn't you want to do this locally?

Reply:
The way the `npm` build command works you could still do versioning locally if you wanted; In Laravel Mix, `mix.config.inProduction` is an indicator if your running `npm run production` or `npm run dev`.

What this enables is `npm run hot` which provided by Laravel Mix, watches for file/css changes and then automatically rebuilds app.js or app.css file when they do change. The issue is if you use mix.version and [npm run hot, you get chunk errors,](https://github.com/JeffreyWay/laravel-mix/issues/568) this conditional fixes that.

So you can still run mix.version() locally by running `npm run production`.

Reply:
Ok thanks for the explanation. Any reason why this isn't in the mix file by default?

Reply:
Nvm. Just read the documentation for it.